arcgis-ios-sdk-samples/Edit data/Edit feature attachments/AttachmentsTableViewController.swift

```diff
@@ -41,53 +40,90 @@ class AttachmentsTableViewController: UITableViewController {
+            guard let self = self else {
+                return
+            }
             if let error = error {
                 print(error)
-            } else {
-                self?.loadAttachments()
+            } else if let attachmentIndex = self.attachments.firstIndex(of: attachment) {
+                let indexPathToRemove = IndexPath(row: attachmentIndex, section: 0)
+                // update the model
+                self.attachments.remove(at: attachmentIndex)
+                // update the table
+                self.tableView.deleteRows(at: [indexPathToRemove], with: .automatic)
             }
         }
     }
     
-    // MARK: - Table view data source
+    private func downloadImage(for attachment: AGSAttachment) {
+        
+        SVProgressHUD.show(withStatus: "Downloading attachment")
+        
+        attachment.fetchData { [weak self] (data: Data?, error: Error?) in
+            
+            SVProgressHUD.dismiss()
+            
+            guard let self = self else {
+                return
+            }
+            
+            if let error = error {
+                print(error)
+            } else if let data = data,
+                let index = self.attachments.firstIndex(of: attachment),
+                let cell = self.tableView.cellForRow(at: IndexPath(row: index, section: 0)) {
+
+                cell.imageView?.image = UIImage(data: data)
+            }
+        }
+    }
+    
+    // MARK: - UITableViewDataSource
     
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         return attachments.count
     }
-    
-    // MARK: - Table view delegate
-    
+
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "AttachmentCell", for: indexPath)
-        
         let attachment = attachments[indexPath.row]
         cell.textLabel?.text = attachment.name
-        
-        cell.imageView?.image = UIImage(named: "CloudDownload", in: AGSBundle(), compatibleWith: nil)
         cell.imageView?.contentMode = .scaleAspectFit
-        cell.imageView?.autoresizingMask = []
-        cell.imageView?.clipsToBounds = true
         if attachment.hasFetchedData {
-            downloadAndSetAttachmentImageForCell(cell, at: indexPath)
+            attachment.fetchData { (data: Data?, error: Error?) in
+                if let data = data {
+                    cell.imageView?.image = UIImage(data: data)
+                }
+            }
+        } else {
+            cell.imageView?.image = UIImage(named: "CloudDownload", in: AGSBundle(), compatibleWith: nil)
         }
         
         return cell
     }
     
+    // MARK: - UITableViewDelegate
+    
+    override func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
+        return !attachments[indexPath.row].hasFetchedData
```

Comment:
👍 Nice! It is still a bit odd that the download button isn't actually a button (usually on the right side of the cell), but that isn't your fault. This is definitely better than it way.